PowerMeta.ps1

```diff
@@ -448,40 +448,41 @@ Function Get-TargetFiles{
 $TASource=@'
   namespace Local.ToolkitExtensions.Net.CertificatePolicy {
     public class TrustAll : System.Net.ICertificatePolicy {
-      public TrustAll() { 
+      public TrustAll() {
       }
       public bool CheckValidationResult(System.Net.ServicePoint sp,
-        System.Security.Cryptography.X509Certificates.X509Certificate cert, 
+        System.Security.Cryptography.X509Certificates.X509Certificate cert,
         System.Net.WebRequest req, int problem) {
         return true;
       }
     }
   }
-'@ 
+'@
           $TAResults=$Provider.CompileAssemblyFromSource($Params,$TASource)
           $TAAssembly=$TAResults.CompiledAssembly
 
           ## We now create an instance of the TrustAll and attach it to the ServicePointManager
           $TrustAll=$TAAssembly.CreateInstance("Local.ToolkitExtensions.Net.CertificatePolicy.TrustAll")
           [System.Net.ServicePointManager]::CertificatePolicy=$TrustAll
-  
+
           ## end code from http://poshcode.org/624
 
-          
+
           $k = 1
           foreach ($link in $targetlinks){
-         
+
           $filename = $link -replace "^.*\/"
           $testpath = ($OutputDir + "\" + $filename)
-          if (!(Test-Path $testpath)) 
+          $outputFileName = $(Split-Path $link -Leaf) -replace '[/:?%~$&{}\\|"]', "_"
+          if (!(Test-Path $testpath))
           {
           Write-Output "Now Downloading $link"
-          Invoke-WebRequest $link -UserAgent $UserAgent -UseBasicParsing -OutFile ($OutputPath + "\" + $(Split-Path $link -Leaf))
+          Invoke-WebRequest $link -UserAgent $UserAgent -UseBasicParsing -OutFile $outputFileName

```

Comment:
This line should be
Invoke-WebRequest $link -UserAgent $UserAgent -UseBasicParsing -OutFile ($OutputPath + "\" + $k + $outputFileName)
as it is it just writes to the base directory